ignite/cmd/chain_init.go

```diff
@@ -105,6 +105,7 @@ func chainInitHandler(cmd *cobra.Command, _ []string) error {
 		chain.KeyringBackend(chaincmd.KeyringBackendTest),
 		chain.WithOutputer(session),
 		chain.CollectEvents(session.EventBus()),
+		chain.CheckCosmosSDKVersion(),
```

Comment:
maybe we can add this check into the chain cmd `PersistentPreRunE` method. wdyt?

https://github.com/ignite/cli/blob/569af5a48ecda21dc3113fff244579d358c25114/ignite/cmd/chain.go#L105-L105

Reply:
That was what I started doing at first but it made more sense to have it as an optional validation option. The main issue with using a pre run handler is that we require the app directory path because it's required to detect the Cosmos SDK version and it's not available at "chain" command level. I think it's possible to do it for each "chain" sub command but the option approach seems simpler and more flexible, in case that validation should eventually be applied outside the context of a command.

At the end I don't have a strong opinion regarding how to implement it so I can change it if you have a strong opinion 😄 

Just one thing to have in mind, if I move it to the pre run handler we will be reading and parsing the `go.mod` file multiple times.

Reply:
Nice, I don't have a strong opinion about it either. It's only because the other migrations are already there. I agree. Let's keep it this way